src/controllers/cartController.js

```diff
@@ -0,0 +1,180 @@
+const Cart = require('../models/Cart');
+const Book = require('../models/Book');
+const { calculateCartTotal } = require('../utils/cartTotal');
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+const { NotFoundError, BadRequestError } = require('../errors');
+const { StatusCodes } = require('http-status-codes');
+
+// get the cart
+const getCart = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+
+    const cart = await Cart.findOne({ createdBy: userId }).populate(
+      'orderItems.book'
+    );
+    if (!cart) {
+      throw new NotFoundError('Cart not found');
+    }
+
+    const updatedItems = cart.orderItems.filter(
+      (item) => item.book && item.book.isAvailable
+    );
+
+    if (updatedItems.length !== cart.orderItems.length) {
+      cart.orderItems = updatedItems;
+      cart.total = calculateCartTotal(cart);
+
+      await cart.save();
+
+      return res.status(StatusCodes.OK).json({
+        message: 'Some unavailable books were removed from your cart.',
```

Comment:
Let's use msg instead of message, since msg is used everywhere in the response when working with books.

Reply:
done